src/shared/icons.utils.js

```diff
@@ -58,6 +63,26 @@ function getTrayIcon(platform, theme) {
 	}
 }
 
+/**
+ * Return partial BrowserWindow options that configure the window icon
+ *
+ * @return {object} partial BrowserWindow constructor options
+ */
+function getBrowserWindowIconOptions() {
+	if (isLinux()) {
+		// There is no standard for setting an (X11) window icon through a desktop file. Instead an
+		// application has to configure it manually. Othwerwise, the default icon of the window
+		// manager will be shown. The icon should preferably be a png.
+		return {
+			icon: icons.window.linux,
+		}
+	}
+
+	// The icon is already handled on other platforms.
+	return {}
+}
```

Comment:
As icons are guaranteed to be only one property `icon`, I'd return the icon path here, not an object with one property, similar to `getTrayIcon`.

It is safe to return `undefined` to use the executable's icon on Mac and Windows.

Reply:
Done